### ISSUE TITLE
Conflict with flutter_test in flutter version 10.2 and built_value_generator

### ISSUE THREAD

**Issue by mclark4386** [google/built_value.dart#528]
Updated flutter and running into a dependency error with `built_value_generator` that is blocking it...

pubspec.yaml
```
name: project_test
description: Test App

dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.2
  http: ^0.12.0
  built_value: ^6.1.5
  built_collection: ^4.0.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  build_runner: any
  built_value_generator: any
```

error:
```
Running "flutter packages get" in project_test...            The current Dart SDK version is 2.1.0-dev.8.0.flutter-bf26f760b1.

Because built_value_generator <5.5.5 requires SDK version >=1.12.0-dev.5.10 <2.0.0 or >=2.0.0-dev <2.0.0 and built_value_generator >=5.5.5 <6.1.5 depends on analyzer ^0.32.1, built_value_generator <6.1.5 requires analyzer ^0.32.1.
And because built_value_generator >=6.1.5 depends on analyzer ^0.33.1 and every version of flutter_test from sdk depends on analyzer 0.33.0, built_value_generator is incompatible with flutter_test from sdk.
So, because project_test depends on both flutter_test any from sdk and built_value_generator any, version solving failed.


pub get failed (1)
Process finished with exit code 1
```

**Comment by davidmorgan** [comment]
Thanks for reporting.

`built_value` needs to move on to the just-released `0.33.3` version of the analyzer, so the right fix for this is for `flutter_test` to also move to that analyzer version.

The `dependency_override` is the right workaround until then.

**Comment by davidmorgan** [comment]
This should now be fixed.
